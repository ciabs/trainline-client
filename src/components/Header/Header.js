import React from 'react';
import {Container, Img, Wrapper} from './Header.styles';
import logo from '../../assets/images/logo.svg';
import {Link} from 'react-router-dom';

const Header = () => {
  return (
    <Container>
      <Wrapper>
        <Link to={'/'}>
          <Img src={logo} alt="Kiwi standing on oval" />
        </Link>
      </Wrapper>
    </Container>
  );
};

export default Header;
