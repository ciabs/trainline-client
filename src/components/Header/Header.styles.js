import styled from 'styled-components';

import colors from '../../styles/colors';
import {media} from '../../helpers/styledComponents.helper';

export const Container = styled.header`
  background-color: ${colors.backgroundColor2};
`;

export const Wrapper = styled.div`
  max-width: 800px;
  margin: auto;
  padding: 15px 0;
  ${media.medium`
    text-align: center;
  `};
`;

export const Img = styled.img`
  height: 40px;
  ${media.medium`
    height: 30px;
  `};
  ${media.small`
    height: 20px;
  `};
`;
