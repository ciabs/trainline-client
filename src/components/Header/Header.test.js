import React from 'react';
import ReactDOM from 'react-dom';
import render from 'react-test-renderer';
import {shallow, configure} from 'enzyme';
import Header from './Header';
import Adapter from 'enzyme-adapter-react-16';
import {Router, BrowserRouter} from 'react-router-dom';

configure({ adapter: new Adapter() });

describe('Header', () => {
  it('should rendere without crashing', () => {
    shallow(<Header />);
  });

  it('should rendere without crashing', () => {
    const div = document.createElement('div');

    ReactDOM.render(
      <BrowserRouter>
        <Header />
      </BrowserRouter>,
      div
    );
  });

  it('should have a valid snapshot', () => {
    const component = render.create(
      <BrowserRouter>
        <Header />
      </BrowserRouter>
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
