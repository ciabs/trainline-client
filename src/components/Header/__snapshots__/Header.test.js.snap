// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should have a valid snapshot 1`] = `
<header
  className="sc-bdVaJa hOPUPK"
>
  <div
    className="sc-bwzfXH qdJsB"
  >
    <a
      href="/"
      onClick={[Function]}
    >
      <img
        alt="Kiwi standing on oval"
        className="sc-htpNat lovHhp"
        src="logo.svg"
      />
    </a>
  </div>
</header>
`;
